libcsm/tests/sls/test_get_hostname.py

```diff
@@ -0,0 +1,77 @@
+#
+#  MIT License
+#
+#  (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+#  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+#  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+#  OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Tests for the sls get_hostname function.
+"""
+
+import http
+import mock
+
+from click.testing import CliRunner
+from libcsm.sls import get_hostname
+
+class MockHTTPResponse:
+    def __init__(self, data, status_code):
+        self.json_data = data
+        self.status_code = status_code
+
+    def json(self):
+        return self.json_data
+
+@mock.patch('kubernetes.config.load_kube_config')
+class TestGetHostname:
+
+    @mock.patch('libcsm.api.Auth', spec=True)
+    @mock.patch('libcsm.sls.api.API.get_management_components_from_sls', spec=True)
+    def test_get_hostname(self, mock_management_components, *_) -> None:
+        """
+        Tests successful run of get_hostname main function.
+        """
+        mock_components = [
+                            {'Parent': 'par1', 'Xname': 'xname1',  'ExtraProperties': {'Aliases': ['ncn-w001'], 'A': 100}},
```

Comment:
nit: whitespace, and add trailing comma

Reply:
canu can catch stuff like this, so we probably need a ticket to add additional linting capabilities

Reply:
I created this, and assigned it to you https://github.com/Cray-HPE/libCSM/issues/46